### PR TITLE
DFPL-479: LA/Admin/Solicitor upload notice of hearing response for placement

### PR DESCRIPTION
**Pull request by DanCatchpole**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFPL-479

### Change description ###
1. Removes the individual fields for notices to be uploaded for LA/Cafcass/Respondent1/2 separately
2. Replaces these fields with a placementNoticeResponses field on the placement class/'complex type' to store these as a list, to allow for any combination of responses and > 2 respondents if needs be
3. Adds an extra option to LA, Admins and External Solicitor's ManageDocuments flow so they can upload a placement notice response _if_ a notice has been issued for a placement application. These will then be visible under the placement tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationCaseField/CareSupervision/la-solicitor.json [hunk 1]
```diff
@@ -1720,6 +1720,27 @@
     "UserRole": "[LASOLICITOR]",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "manageDocumentsPlacementList",
+    "UserRole": "[LASOLICITOR]",
+    "CRUD": "CRU"
```
Comment (jamesrferguson1) [inline ccd-definition/AuthorisationCaseField/CareSupervision/la-solicitor.json:1728]: Are admins only allowed to delete?

File: ccd-definition/CaseEventToComplexTypes/manageDocs/manageDocumentsSolicitorNoticeDocument.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[
+  {
```
Comment (jamesrferguson1) [inline ccd-definition/CaseEventToComplexTypes/manageDocs/manageDocumentsSolicitorNoticeDocument.json:2]: Do we need this? (As the same as the admin)

File: ccd-definition/CaseEventToFields/CareSupervision/manageOrders/2_orderCreateSection.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
     "CaseTypeID": "CARE_SUPERVISION_EPO",
     "CaseEventID": "manageOrders",
     "CaseFieldID": "manageOrdersUploadTypeOtherTitle",
```
Comment (jamesrferguson1) [inline ccd-definition/CaseEventToFields/CareSupervision/manageOrders/2_orderCreateSection.json:73]: Check whether we need to do this

File: ccd-definition/CaseEventToFields/CareSupervision/manageOrders/5_orderDetailsSection.json [hunk 4]
```diff
@@ -543,7 +543,7 @@
     "PageID": "OrderDetails",
     "PageDisplayOrder": 6,
     "PageColumnNumber": 1,
```
Comment (jamesrferguson1) [inline ccd-definition/CaseEventToFields/CareSupervision/manageOrders/5_orderDetailsSection.json:545]: Also double check why this is here

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsLAControllerAboutToSubmitTest.java [hunk 5]
```diff
@@ -424,6 +428,75 @@ void shouldPopulateRespondentStatementsCollection() {
...
+        CaseData responseData = extractCaseData(postAboutToSubmitEvent(caseData, USER_ROLES));
+        assertThat(responseData.getPlacementEventData().getPlacements().get(0).getValue().getNoticeDocuments())
+            .hasSize(1);
+
+        assertExpectedFieldsAreRemoved(responseData);
+    }
+
+    @Test
+    void shouldNotOverrideExistingCafcassRespondentNotices() {
+        // TODO - add test similar to solicitor where cafcass + respondent notices are already present
```
Comment (jamesrferguson1) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsLAControllerAboutToSubmitTest.java:458]: Remove the TODO

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsLAControllerMidEventTest.java [hunk 6]
```diff
@@ -100,6 +107,7 @@ void shouldInitialiseHearingList() {
             .type(FURTHER_EVIDENCE_DOCUMENTS)
             .hasHearings("Yes")
```
Comment (jamesrferguson1) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsLAControllerMidEventTest.java:108]: Use the enums

**Comment by DanCatchpole** [comment]
superseded by #3651

**Pull request closed without merging**